File: android/app/src/main/java/at/area23/schnapslet/models/TwoCards.java
/*
 *
 * @author           Heinrich Elsigan root@darkstar.work
 * @version          V 1.6.9
 * @since            API 26 Android Oreo 8.1
 *
 */
/*
   Copyright (C) 2000 - 2023 Heinrich Elsigan root@darkstar.work

   Schnapslet java applet is free software; you can redistribute it and/or
   modify it under the terms of the GNU Library General Public License as
   published by the Free Software Foundation; either version 2 of the
   License, or (at your option) any later version.
   See the GNU Library General Public License for more details.

*/
package at.area23.schnapslet.models;

import at.area23.schnapslet.constenum.CARDCOLOR;

/**
 * TwoCards class represents a pair (marriage) of cards.
 *
 * @see <a href="https://github.com/heinrichelsigan/schnapslet/wiki</a>
 */
public class TwoCards {
    
	// boolean atou = false;
	// CARDCOLOR cardColor = CARDCOLOR.NONE; // Color of Pair
	// char color = 'n';   // 4 colors and 'n' for unitialized
    // int pairValue = 20;      // 5 values and 0 for unitialized
	public Card[] cards = new Card[2]; // the 2 Cards, that represents the pair
	public Card card1st = null;
	public Card card2nd = null;
	public int cardSumValue = 0;

    /**
     * Constructor TwoCards()
     */
    public TwoCards() {
        super();
		card1st = null;
		card2nd = null;
		cards = new Card[2];
		cards[0] = card1st;
		cards[1] = card2nd;
    }

	/**
	 * Constructor PairCard
	 * @param first represents 1st card
	 * @param second represents 2nd card
	 */
	public TwoCards(Card first, Card second) {
		this();
		
		card1st = new Card(first);
		card2nd = new Card(second);

		cards = new Card[2];
		cards[0] = card1st;
		cards[1] = card2nd;

		this.cardSumValue = card1st.getValue() + card2nd.getValue();
	}

}

File: asp.net/SchnapsNet/Models/TwoCards.cs
using SchnapsNet.ConstEnum;
using System;
using System.Collections.Generic;
using System.Drawing;
using System.IO;
using System.Linq;
using System.Resources;
using System.Runtime.Remoting.Contexts;
using System.Security.Policy;
using System.Web;
using System.Web.UI.WebControls;

namespace SchnapsNet.Models
{
	/// <summary>
	/// Port of class TwoCards
	/// </summary>
	public class TwoCards
    {
        private Card[] cards = new Card[2]; // the 2 Cards, that represents the pair
        private Card card1st = null;
        private Card card2nd = null;
        internal int cardSumValue = 0;

        #region properties
        
        /// <summary>
        /// Pair of cards
        /// </summary>
        public Card[] Cards { get => cards; }
        /// <summary>
        /// 1st card of pair
        /// </summary>
        public Card Card1st { get => card1st; protected set => card1st = value; }
        /// <summary>
        /// 2nd card of pair
        /// </summary>
        public Card Card2nd { get => card2nd; protected set => card2nd = value; }

        #endregion properties

        #region ctor
        /// <summary>
        /// Constructor TwoCards()
        /// </summary>
        public TwoCards()
        {
            card1st = null;
            card2nd = null;
            cards = new Card[2];
            cards[0] = card1st;
            cards[1] = card2nd;
        }   
        
        /// <summary>
        /// Constructor with two Cards as agrument
        /// </summary>
        /// <param name="first">first represents 1st card</param>
        /// <param name="second">represents 2nd card</param>
        public TwoCards(Card first, Card second) : this()
        {
            card1st = new Card(first);
            card2nd = new Card(second);

            cards = new Card[2];
            cards[0] = card1st;
            cards[1] = card2nd;

            this.cardSumValue = card1st.CardValue.GetValue() + card2nd.CardValue.GetValue();
        }
        #endregion ctor

    }
}